### DISCUSSION TITLE
Option to have only the opposite shift work

### DISCUSSION THREAD

**Discussion by ghost** [monkeytypegame/monkeytype#1021] [Ideas] [answered]
i.e., an option to make only the shift opposite to the letter being typed uppercase that key. Just so it can be easier to master which shift to press.

**Comment by frogfrogfrogfrogfrog** [comment]
That seems kinda useless considering that you don't really need to use opposite shift keys to type fast or to balance out the key usage on your hands. You could also just do this yourself right? That way you can develop your own muscle memory and even optimize which shift keys feels better on what letters. Like for the letter "b" or "y" you can develop your own muscle memory.

But if @Miodec feels like this is a helpful feature I'm not gonna try to stop him.

**Reply by ghost** [reply]
I'd rather we have it, it won't harm to have the option to do it the 'correct' way, plus it makes a difference, at least for the left most and right most keys, where if you habitually use the same-side shift, you'd use the ring finger instead of the pinky. But really it is up to @Miodec

**Comment by Miodec** [comment, marked as answer]
Couple of problems i can see straight away:
 - How do I know what side of the keyboard the keys are?
 - How do I know which layout the user is using? There is no way of detecting the physical layout

Even if I assumed this only works if you use layout override, it breaks other things like dead keys and i don't know if you can block shift events depending on of its left or right side anyway.

**Reply by ghost** [reply]
I think this works regardless of layout: https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/code
https://stackoverflow.com/questions/22029033/can-javascript-tell-the-difference-between-left-and-right-shift-key

But even if this doesn't work, just make it work for the most common use case, and if someone complains then update the code.

**Reply by Miodec** [reply]
Yeah, but it *does* need to work depending on the layout because if a key changes sides from left to right it will require the use of the opposite shift key.

**Reply by ghost** [reply]
I get you, but now that there is a method that does work, can you add support? I am assuming it works, because this MDN link has a "Try it out" which, not only detects the appropriate shift key, but also detects the original key even when I change the layout or the language.

**Reply by Miodec** [reply]
Right, i see - `event.code` ignores keyboard layout. This makes kinda possible.

EDIT: Added #1059
